core/src/main/java/com/amplifyframework/storage/StorageServerSideEncryption.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.storage;
+
+/**
+ * Represents storage server side encryption method type.
+ */
+public enum StorageServerSideEncryption {
+    /**
+     * AES256 encryption.
+     */
+    MANAGED_KEYS("AES256"),
+
+    /**
+     * aws:kms encryption.
+     */
+    KMS_KEYS("aws:kms"),
```

Comment:
Is this too AWS specific to be in `core` module?

At Amplify layer, the categories should not adhere too much to a specific AWS service, and such details should be specified at the plugin layer. Exposing `AES256` and `KMS` as supported encryption methods at a category level doesn't seem ideal to me, although I guess one can still argue that these are generic enough encryptions to be applicable to storage?

Reply:
Well, this is used across multiple modules, so I did not find a better place to put it.
Also, on the AWSMobileClient only these encryption methods are specified. I think this is a pretty required feature.

Reply:
I believe the ideal place for `StorageServerSideEncryption` enum to be housed in is inside `aws-storage-s3` module. However, the way that `StorageOptions` is structured makes it difficult to pass in plugin-specific options parameter to the category. This may require some refactoring from our side.

Meanwhile, this PR looks pretty solid otherwise. As you mentioned, the enum in question cannot be placed outside of `core` as it currently stands, so I think we will do a quick refactor of `StorageOptions` after merging in your changes.  How does that sound to you?

Reply:
@raphkim sounds great! <3